src/App.tsx

```diff
@@ -1,24 +1,153 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import { getTodos, addTodo } from './api/todos';
+import { TodoList } from './components/TodoList';
+import { useTodoContext } from './context';
+import { Todo } from './types/Todo';
 
-const USER_ID = 0;
+const USER_ID = 12113;
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState<string>('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const {
+    setAllTodos,
+    visibleTodos,
+    errorHandler,
+    errorMessage,
+    setErrorMessage,
+    inputRef,
+  } = useTodoContext();
+
+  const handleSubmit = async (event: React.FormEvent) => {
+    event.preventDefault();
+    const trimmedTitle = query.trim();
+
+    if (trimmedTitle === '') {
+      errorHandler('Title should not be empty');
```

Comment:
I think you should use an enum for Error, for example:
```
export enum ErrorType {
  Load = 'Unable to load todos',
  Title = 'Title should not be empty',
  ...
}

if (trimmedTitle === '') {
      errorHandler(ErrorType.title);
...
```

Reply:
true